deploy/agent-vm.mk

```diff
@@ -3,9 +3,27 @@ VMRAM ?= 512
 VMCPUS ?= 1
 VMDISK = /var/lib/libvirt/images/$(VMNAME).qcow2
 VMWAIT ?= 0
+IP=$(shell test/scripts/get_ext_ip.sh)
 
-bin/output/qcow2/disk.qcow2: hack/Containerfile.local bin/.rpm
-	sudo podman build -f hack/Containerfile.local -t localhost/local-flightctl-agent:latest .
+
+out/.agent-bootc-images: out rpm hack/agent-image/Containerfile-e2e-base.local hack/agent-image/Containerfile-e2e-v2.local hack/agent-image/Containerfile-e2e-v3.local
```

Comment:
I am getting nervous about the complexity of these make files. Is there anything we can do to make this easier for folks to grok.

Like what does this mean to a developer who is trying to understand a failing test.

Reply:
About the makefile complexity... yes, it's getting complex. Maybe we should pack some of those things in simpler sell scripts, as we do for the helm deploy, etc. Also, some shell script would help make those things leaner. I'll apply that idea to this patch more and see how it looks.


To understand a failing test, I think it's good to write idiomatic and very well-documented tests so it's clear what the intent of the test is, what images are involved, and what those images contain. I should probably add a section describing the 3 images we have and what is in each.

Reply:
Sure but if the test is failing because of test infra or someone is trying to extended e2e they end up here. Thanks for looking into it.